src/irc/core/irc-servers.c

```diff
@@ -409,7 +423,10 @@ static void sig_disconnected(IRC_SERVER_REC *server)
                         server_redirect_destroy(tmp->next->data);
 	}
 	g_slist_free(server->cmdqueue);
-        server->cmdqueue = NULL;
+	server->cmdqueue = NULL;
+
+	g_slist_free_full(server->cap, g_free);
```

Comment:
I don't know what's the current glib version requirement of irssi (configure.ac says 2.6.0 but that's obviously not true) but `g_slist_free_full()` was added in 2.28